util/image_viewer/spotfinder_frame.py

```diff
@@ -1652,7 +1652,14 @@ def map_coords(x, y, p):
                             raise
                     beam_x, beam_y = detector[panel].millimeter_to_pixel(beam_centre)
                     beam_x, beam_y = map_coords(beam_x, beam_y, panel)
-                    for i, h in enumerate(((10, 0, 0), (0, 10, 0), (0, 0, 10))):
+                    self.params.basis_vector_scale
+                    for i, h in enumerate(
+                        (
+                            (self.params.basis_vector_scale, 0, 0),
+                            (0, self.params.basis_vector_scale, 0),
+                            (0, 0, self.params.basis_vector_scale),
+                        )
+                    ):
```

Comment:
```suggestion
                    for i, h in enumerate(((1, 0, 0), (0, 1, 0), (0, 0, 1))):
                        r = A * matrix.col(h) * self.params.basis_vector_scale
```
would probably achieve the same in fewer lines of code (and probably be more readable).